app/src/test/java/org/mozilla/fenix/library/bookmarks/BookmarkControllerTest.kt

```diff
@@ -101,11 +102,25 @@ class BookmarkControllerTest {
 
         verifyOrder {
             invokePendingDeletion.invoke()
-            homeActivity.browsingModeManager.mode = BrowsingMode.Normal
             homeActivity.openToBrowserAndLoad(item.url!!, true, BrowserDirection.FromBookmarks)
         }
     }
 
+    @Test
+    fun `handleBookmarkTapped should respect browsing mode`() {
+        // if in normal mode, should be in normal mode
+        every { homeActivity.browsingModeManager.mode } returns BrowsingMode.Normal
+
+        controller.handleBookmarkTapped(item)
+        Assert.assertEquals(BrowsingMode.Normal.isPrivate, homeActivity.browsingModeManager.mode.isPrivate)
```

Comment:
nit:
```suggestion
        assertEquals(BrowsingMode.Normal, homeActivity.browsingModeManager.mode)
```

Reply:
yeah, this just seems so obvious, no clue why i consumed the `isPrivate` thing